frontend/src/components/layout/SideNavBar/NavIcon.tsx

```diff
@@ -0,0 +1,66 @@
+import clsx from 'classnames';
+import TooltipWrapper from 'components/common/TooltipWrapper';
+import { Roles } from 'constants/index';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import _ from 'lodash';
+import { Nav } from 'react-bootstrap';
+import styled from 'styled-components';
+
+interface INavIconProps {
+  icon: React.ReactElement;
+  text: string;
+  showText: boolean;
+  onClick: () => void;
+  roles?: Roles[];
+}
+
+/**
+ * Component that creates a nav, with an icon, and optional text.
+ * @param {INavIconProps} param0
+ */
+export const NavIcon = ({ icon, text, showText, onClick, roles }: INavIconProps) => {
+  const { hasRole } = useKeycloakWrapper();
+  const displayIcon = _.find(roles, (role: Roles) => hasRole(role));
+
+  return !roles?.length || displayIcon ? (
+    <StyledNav onClick={onClick} data-testid={`nav-tooltip-${text.replace(' ', '').toLowerCase()}`}>
+      <StyledLink>
+        <TooltipWrapper toolTipId={`nav-tooltip-${text}`} toolTip={text}>
+          {icon}
+        </TooltipWrapper>
+        <StyledLabel className={clsx({ show: showText })}>{text}</StyledLabel>
+      </StyledLink>
+    </StyledNav>
+  ) : null;
+};
+
+const StyledNav = styled(Nav.Item)`
+  width: 100%;
+`;
+
+const StyledLink = styled(Nav.Link)`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  svg {
+    min-width: max-content;
+  }
+`;
+
+const StyledLabel = styled.label`
```

Comment:
Wonder if we should start separating these into their own files as discussed last week.

Reply:
you wonder this a lot apparently.